appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/MessageBeanInterfaceGenerator.java

```diff
@@ -16,27 +17,29 @@
 
 package org.glassfish.ejb.mdb;
 
-import com.sun.ejb.codegen.EjbOptionalIntfGenerator;
-
 import jakarta.resource.spi.endpoint.MessageEndpoint;
 
+import com.sun.ejb.codegen.EjbOptionalIntfGenerator;
+
 public class MessageBeanInterfaceGenerator extends EjbOptionalIntfGenerator {
 
     public MessageBeanInterfaceGenerator(ClassLoader loader) {
         super(loader);
     }
 
-    public Class generateMessageBeanSubClass(Class<?> beanClass, Class messageBeanInterface) throws Exception {
```

Comment:
Could we limit it somehow, like specify <T> Class<? extends T> gmbs(Class<T> . Not sure, if it is not possible, let it be, it is just an idea.